### PR TITLE
Upgrade to repo-config v0.6.1

### PR DESCRIPTION
**Pull request by llucax**
The new repo-config version brings a few important changes:

* Dependabot grouping
* Replace `darglint` with `pydoclint`
* Introduction of `flake8` basic checks
* Start checking docstrings and `flake8` basic checks in tests and benchmarks

Becuase of this, this PR is very big, as it needs to fix a lot of issues detected by `pydoclint` and `flake8`.

The first commits, up to **Add `mypy` configuration to `pyproject.toml`** deal with the upgrade of repo-config templates.

The rest of the commits are documentation fixes (first from `pydoclint` and then from `pydocstyle`) plus some other `flake8` fixes, usually in that order.

### CODE REVIEW

**Comment by llucax** [comment]
OK, this is working and only waiting for the repo-config release. It would be good to merge ASAP so new PRs don't introduce new `flake8` failures and this needs to be updated ad-infinitum.

**Comment by llucax** [comment]
Nice side effect: this also speeds up the CI by a minute or two.

**Comment by llucax** [comment]
Ready for review, updated branch protection rules for the new matrix variable, enabled auto-merge.

File: src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_grid_power_formula.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
 class GridPowerFormula(FormulaGenerator[Power]):
     """Creates a formula engine from the component graph for calculating grid power."""
 
-    def generate(
+    def generate(  # noqa: DOC502
+        # * ComponentNotFound is raised indirectly by _get_grid_component_successors
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_grid_power_formula.py:16]:
```suggestion
        # ComponentNotFound is raised indirectly by _get_grid_component_successors
```

File: src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_pv_power_formula.py [hunk 2]
```diff
@@ -17,14 +17,19 @@
 class PVPowerFormula(FormulaGenerator[Power]):
     """Creates a formula engine for calculating the PV power production."""
 
-    def generate(self) -> FormulaEngine[Power]:
+    def generate(  # noqa: DOC502 (ComponentNotFound and RuntimeError are raised
+        #                          indirectly by _get_pv_power_components)
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_pv_power_formula.py]: Feel free to discard this one
```suggestion
    def generate(  # noqa: DOC502 
        # * ComponentNotFound is raised indirectly by _get_pv_power_components
        # * RuntimeError is also raised indirectly by _get_pv_power_components
```

**Review by tiyash-basu-frequenz — DISMISSED** [review]

File: src/frequenz/sdk/timeseries/_quantities.py [hunk 3]
```diff
@@ -563,6 +566,9 @@ def __truediv__(self, other: Current) -> Voltage:
 
         Args:
             other: The current to divide by.
+
+        Returns:
+            A voltage from dividing this power by the a current.
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/_quantities.py:571]:
```suggestion
            The voltage from dividing this power by the current.
```

File: src/frequenz/sdk/timeseries/_quantities.py [hunk 4]
```diff
@@ -571,6 +577,9 @@ def __truediv__(self, other: Voltage) -> Current:
 
         Args:
             other: The voltage to divide by.
+
+        Returns:
+            A current from dividing this power by a voltage.
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/_quantities.py:582]:
```suggestion
            The current from dividing this power by the voltage.
```

File: src/frequenz/sdk/timeseries/_quantities.py [hunk 5]
```diff
@@ -903,6 +918,9 @@ def __truediv__(self, other: timedelta) -> Power:
 
         Args:
             other: The duration to divide by.
+
+        Returns:
+            A power from dividing this energy by the given duration.
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/_quantities.py:923]:
```suggestion
            The power from dividing this energy by the given duration.
```

File: src/frequenz/sdk/timeseries/_quantities.py [hunk 6]
```diff
@@ -911,6 +929,9 @@ def __truediv__(self, other: Power) -> timedelta:
 
         Args:
             other: The power to divide by.
+
+        Returns:
+            A duration from dividing this energy by the given power.
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/_quantities.py:934]:
```suggestion
            The duration from dividing this energy by the given power.
```

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I only found a few cosmetics to check for

File: src/frequenz/sdk/timeseries/_quantities.py (hunk 3 above)
Comment (llucax) [reply src/frequenz/sdk/timeseries/_quantities.py:571]: These are actually not that well phrased either, it should be `A/The voltage **resulting** from dividing..." or something like that. Maybe we should just use `The calculated voltage` as it is a bit redundant with the method description (which is also wrongly phrased).

I think I will leave these comments for a followup PR improving docs.

File: src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_grid_power_formula.py (hunk 1 above)
Comment (llucax) [reply src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_grid_power_formula.py:16]: I prefer to keep the consistent formatting with one item per exception, even if we have only one.

Comment (daniel-zullo-frequenz) [reply src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_grid_power_formula.py:16]: I see. There was a case above when it was used without the `*` and that's why I suggested it here

**Comment by llucax** [comment]
@daniel-zullo-frequenz I just replied to your comments. If it's OK with you please approve so it can be merged.

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Review by daniel-zullo-frequenz — DISMISSED** [review]

**Review by daniel-zullo-frequenz — APPROVED** [review]

File: src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_grid_power_formula.py (hunk 1 above)
Comment (llucax) [reply src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_grid_power_formula.py:16]: Oh, OK. I did them at different times and had to adjust them to fulfill some new checks by `flake8`, that's why there probably were a couple of mixed styles

**Pull request merged**